Pattern Grid: Update design of pattern cards

ryelle opened the pull request:
Fixes #103 - This updates the UI of the pattern cards to match the new design:
- Title & meta are always visible, only adding favorite count for now since we don't have views
- Show small author avatar
- Show a status label on draft/pending/declined patterns (seen in My Patterns)
- Fix an issue with encoded characters in pattern titles (ex, `&amp;`)
- Fix extra space in the preview — should fix "Extra space on the left/right of full-width items" in #156 

❗  This uses a fake favorite value for now, to get the UI working. I think we'll need a solution with caching to get this value for real, so that discussion can happen in #100 

### Screenshots

Browsing patterns:
![all-patterns](https://user-images.githubusercontent.com/541093/121741417-82b57000-cacc-11eb-96d7-ca3e904bbd2e.png)

My patterns:
![my-patterns](https://user-images.githubusercontent.com/541093/121741418-83e69d00-cacc-11eb-8e3b-d194a3da1eca.png)

Responsive styles - drops to 2 column around 770px, and to 1 col around 525px.
![all-patterns-small](https://user-images.githubusercontent.com/541093/121741414-81844300-cacc-11eb-8b07-6d9a0fbe5ce8.png)

![all-patterns-mobile](https://user-images.githubusercontent.com/541093/121741410-80ebac80-cacc-11eb-8f30-878222796f59.png)


### How to test the changes in this Pull Request:

- Build the branch with `yarn workspaces run build`
- Make sure you have a page called My Patterns (see #152), and a handful of patterns by your user
- View the pattern archives (all and categories)
- They should match the figma design
- View My Patterns
- The patterns should match the design, draft & pending patterns should be labelled
- Resizing the screen shouldn't break anything

ryelle committed:
Pattern Grid: Always show title and meta

ryelle committed:
Decode HTML entities in rendered title

ryelle committed:
Add a (temporary) random favorite count for testing

ryelle committed:
Pass down render function to PatternGrid to better control PatternThumbnail

ryelle committed:
Add the status label

ryelle committed:
Fix avatar & view alignment

ryelle committed:
Update colors, fix button style

ryelle committed:
Improve display on smaller screens, fix spacing at footer

StevenDufresne commented on public_html/wp-content/themes/pattern-directory/src/components/iframe/index.js:
```diff
@@ -93,14 +93,27 @@ function setHead( doc, head ) {
 	} else {
 		headHTML +=
-			'<link rel="stylesheet" href="https://wp-themes.com/wp-content/themes/twentytwentyone/style.css" media="all" />';
+			'<link rel="stylesheet" href="https://wp-themes.com/wp-content/themes/twentynineteen/style.css" media="all" />';
```
This isn't mentioned in the description. Is this on purpose?

StevenDufresne replied:
I now see this mentioned in #177.

StevenDufresne commented on public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/index.js:
```diff
@@ -1,29 +1,65 @@
+
+			<h2 className="pattern-grid__title">{ decodeEntities( pattern.title.rendered ) }</h2>
+			<p className="pattern-grid__meta">
+				{ showAvatar && pattern.author_avatar ? (
```
Should there be a default case for missing authors... if that case is possible??

StevenDufresne commented on public_html/wp-content/themes/pattern-directory/src/components/pattern-thumbnail/index.js:
```diff
@@ -1,29 +1,65 @@
-				<CopyPatternButton isSmall={ true } content={ pattern.pattern_content } />
+			<div className="pattern-grid__pattern-frame">
+				<a href={ pattern.link } rel="bookmark">
+					<span className="screen-reader-text">{ decodeEntities( pattern.title.rendered ) }</span>
```
I can't remember, but are we relying on `.org` to have this style globally?

ryelle replied:
Do you mean the `screen-reader-text`? it's included in [the parent theme's CSS](https://meta.trac.wordpress.org/browser/sites/trunk/wordpress.org/public_html/wp-content/themes/pub/wporg/css/objects/_accessibility.scss), so it's bundled into the built styles.

StevenDufresne replied:
👍

StevenDufresne reviewed (approved):
A few minor comments. Looks good.

shaunandrews commented:
I think we should link the pattern title to view the pattern. Similarly, the author's avatar should be linked to their -archive page.

--

It'd be nice if you could click the heart icon to favorite the pattern. I know its a duplicate action, but it feels strange to have the icon _not_ be interactive.

--

> only adding favorite count for now since we don't have views

Will we have views? If not, perhaps we can display the authors name?

--

<img width="999" alt="image" src="https://user-images.githubusercontent.com/191598/121959663-112b2b00-cd33-11eb-9e42-62a836edb324.png">

I'm seeing teeny-tiny scrollbars, and many patterns are rendered outside their container — even when viewing on the detail page.

--

<img width="152" alt="image" src="https://user-images.githubusercontent.com/191598/121959855-52bbd600-cd33-11eb-9f1d-133af91f0520.png">

The Favorite button's `:focus` state is a single pixel line without any border-radius. Instead, it should match the copy button's focus state, using a thicker border with the border-radius.

--

On My Patterns I'm able to see a "Draft" indicator:

<img width="110" alt="image" src="https://user-images.githubusercontent.com/191598/121960026-90206380-cd33-11eb-8d24-c8fc221c1011.png">

In the designs, the text appears bolder and there's a border-radius applied to the shape:

<img width="122" alt="image" src="https://user-images.githubusercontent.com/191598/121960103-a8907e00-cd33-11eb-9b4b-173a4dc8834e.png">

Also, maybe the badges should have `pointer-events: none;` to that it doesn't block the ability to click the pattern or allow people to select it's text.

shaunandrews commented:
I've still got this branch running and noticed that the copy button has a nice, subtle hover state — but the favorite button doesn't. Can we add a hover state for the favorite button:

<img width="355" alt="image" src="https://user-images.githubusercontent.com/191598/122115472-86116a00-cdf2-11eb-952d-d29ba2e2d139.png">

ryelle committed:
Display pattern author name

ryelle committed:
Hide the scrollbars on the small preview frames

ryelle committed:
Add links to author & post title

ryelle committed:
Make small favorite count a working favorite button

ryelle committed:
Fix style on status tag

ryelle committed:
Give favorite button a slightly better label for screen reader users

ryelle commented:
Okay, latest changes include:
- Author display name is shown, and links to the author page (author filter doesn't work yet though)
- Title links to pattern
- Favorite count is now also a favorite button
- Removed the scrollbars from the small previews
- Fixed the style on the status labels, no longer block clicks

A favorited pattern with favorite focused:
<img width="325" alt="Screen Shot 2021-06-16 at 5 20 32 PM" src="https://user-images.githubusercontent.com/541093/122295423-2afd7700-cec7-11eb-8b17-9e69014075a9.png">

A pending pattern with favorite hovered:
<img width="333" alt="Screen Shot 2021-06-16 at 5 20 04 PM" src="https://user-images.githubusercontent.com/541093/122295426-2c2ea400-cec7-11eb-9310-483d44298277.png">

^ The above don't have authors because it's the My Patterns page. From the main archive:
<img width="990" alt="Screen Shot 2021-06-16 at 5 19 00 PM" src="https://user-images.githubusercontent.com/541093/122296281-39985e00-cec8-11eb-87fb-9e58d1b7c805.png">

ryelle committed:
Avoid using global, which may be undefined

ryelle commented:
Merging this, any followup feedback can be addressed in issues.
